extensions/event-api/src/main/java/io/opentelemetry/extension/event/EventLogger.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.event;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.logs.Logger;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+
+/**
+ * {@link EventLogger} provides convenience methods for emitting Events. An Event is a {@code
+ * LogRecord} that conform to the <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/semantic_conventions/events.md">Event
+ * Semantic Conventions</a>.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * class MyClass {
+ *   private final LoggerProvider loggerProvider = GlobalLoggerProvider.get();
+ *   // Create EventLogger by wrapping Logger and setting event domain to "acme-observability"
+ *   private final EventLogger eventLogger =
+ *     EventLogger.create(
+ *       loggerProvider.get("instrumentation-library-name"), "acme.observability");
+ *
+ *   void doWork() {
+ *     eventLogger.emitEvent("my-event", Attributes.builder()
+ *          .put("key1", "value1")
+ *          .put("key2", "value2").build())
+ *   }
+ * }
+ * }</pre>
+ */
+public final class EventLogger {
```

Comment:
During the java SIG meeting, it was widely agreed that this name is not great, and that `EventEmitter` might be preferable.

Reply:
Perhaps this should be made an interface.

Reply:
> Perhaps this should be made an interface.

That suggests alternative implementations. The current was to use one of these is something like:

```java
LoggerProvider loggerProvider = ... // Configure a logger provider
Logger logger = loggerProvider.get("my-logger");
EventEmitter eventEmitter = EventLogger.create(logger, "my-event-domain");
```

What kind of workflow did you have in mind that would allow for alternative implementations to be plugged in?

Reply:
I know that we are far away from implementing _events_ independently from _logs_, and this might even never happen. But I don't agree that alternative implementations must exist in order to define an interface.
I hope that a new interface will help emphasize that _events_ are conceptually different than _logs_.
And the new interface will at least allow us to provide a different implementation in the future without any breaking changes - even if this is hypothetical at this point.